src/v/storage/disk_log_impl.cc

```diff
@@ -150,12 +151,28 @@ model::offset disk_log_impl::time_based_gc_max_offset(model::timestamp time) {
     // files so that size-based retention eventually evicts the problematic
     // segment, preventing a crash.
     //
+
+    // if the segment max timestamp is bigger than now plus threashold we
+    // will report the segment max timestamp as bogus timestamp
+    static constexpr auto const_threshold = 1min;
+    auto bogus_threshold = model::timestamp(
+      model::timestamp::now().value() + const_threshold / 1ms);
```

Comment:
`std::chrono::time_point + std::chrono::duration` should work, we shouldn't sidestep the type system.

Reply:
the problem is that `threshold` is not std::chrono::...::time_point` type. it is our custom timestamp type

Reply:
Oh, yeah.